core/kernel/notif.c

```diff
@@ -101,7 +102,7 @@ void notif_send_async(uint32_t value)
 
 	DMSG("0x%"PRIx32, value);
 	bit_set(notif_values, value);
-	itr_raise_pi(CFG_CORE_ASYNC_NOTIF_GIC_INTID);
+	itr_chip->ops->raise_pi(itr_chip, CFG_CORE_ASYNC_NOTIF_GIC_INTID);
```

Comment:
How come a wrapper is missing for this function pointer?

Reply:
There is no return value, so failure would panic in any way.
Maybe  a test in `notif_alloc_async_value()` to return NOT_SUPPORTED is there's no `.raise_pi` hander.

Reply:
My concern was rather that we're setting an example to reach directly into the ops struct. If that's OK, why do we have `interrupt_enable()` for instance?

Reply:
I think it is better to have consumers using `interrupt_xxx()` helper function to abstract how the framework implements the resource.

I can add inline `interrut_raise_pi()` and like helper functions.

Reply:
OK, please do.